capstone-app/src/app/modules/maps/map/map.component.spec.ts

```diff
@@ -18,14 +19,64 @@ describe('MapComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(MapComponent);
-    component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
-
   it('should create the app', () => {
     // const fixture = TestBed.createComponent(MapComponent);
     // const app = fixture.componentInstance;
     expect(fixture.componentInstance).toBeTruthy();
   });
+
+  it('test map marker click event', () => {
+    const fixture = TestBed.createComponent(MapComponent);
+    fixture.detectChanges();
+
+    const coordinates = new google.maps.LatLng({lat: 26.011760, lng:  -80.139050});
+
+    fixture.componentInstance.markers.push(coordinates);
+    fixture.detectChanges();
+
+    const markerDebug = fixture.debugElement.query(By.css('.marker'));
+    const markerComponent = markerDebug.componentInstance;
+
+    expect(markerComponent.getPosition()).toEqual(coordinates);
+  });
+
+  it('test multiple map marker click event', () => {
+    const fixture = TestBed.createComponent(MapComponent);
+    fixture.detectChanges();
+    let coordinates: google.maps.LatLng;
+    let lat = 26.011760;
+    let lng = -80.139050
+    for (let i = 0; i < 5; i++) {
+      coordinates = new google.maps.LatLng(lat, lng);
+      lat++;
+      lng++;
+      fixture.componentInstance.markers.push(coordinates);
+      fixture.detectChanges();
+    }
+
+    const markerDebug = fixture.debugElement.queryAll(By.css('.marker'));
+    const markerComponent = markerDebug.map(marker => marker.componentInstance.getPosition());
+
+    expect(markerComponent).toEqual(fixture.componentInstance.markers);
+  });
+
+
+  it('sets center and zoom of the map', () => {
+    const options = {center: {lat: 26.011760, lng:  -80.139050}, zoom: 13};
+
+    const fixture = TestBed.createComponent(MapComponent);
+    fixture.componentInstance.location = new google.maps.LatLng(options.center);
+    fixture.componentInstance.zoom = options.zoom;
+    fixture.detectChanges();
+
+    fixture.componentInstance.location = new google.maps.LatLng({lat: 4.624335, lng: -74.063644});
+    fixture.componentInstance.zoom = 12;
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.location.toJSON()).toEqual({lat: 4.624335, lng: -74.063644});
```

Comment:
This doesn't really test anything since we know that we've already set zoom and location to these values. Instead, get a reference to the google map component (similar to how you got references to the markers), and see if the zoom and center values it is getting matches the ones you're setting on your component.